File: 01. Crystal Resonance/CrystalResonance.java
import java.util.*;

public class CrystalResonance {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            int n = sc.nextInt();
            int[] powers = new int[n];

            for (int i = 0; i < n; i++) {
                powers[i] = sc.nextInt();
            }

            if (hasDuplicate(powers)) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }

        sc.close();
    }

    public static boolean hasDuplicate(int[] arr) {
        Set<Integer> seen = new HashSet<>();
        for (int power : arr) {
            if (seen.contains(power)) {
                return true;
            }
            seen.add(power);
        }
        return false;
    }

}

File: 01. CrystalResonance/CrystalResonance.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

bool hasDuplicate(const vector<int> &arr)
{
    unordered_set<int> seen;
    for (int power : arr)
    {
        if (seen.find(power) != seen.end())
        {
            return true;
        }
        seen.insert(power);
    }
    return false;
}

int main()
{
    int t;
    cin >> t;

    while (t--)
    {
        int n;
        cin >> n;
        vector<int> powers(n);

        for (int i = 0; i < n; i++)
        {
            cin >> powers[i];
        }

        if (hasDuplicate(powers))
        {
            cout << "YES" << endl;
        }
        else
        {
            cout << "NO" << endl;
        }
    }

    return 0;
}